packages/dynamic_layouts/lib/src/wrap_layout.dart

```diff
@@ -0,0 +1,271 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/rendering.dart';
+
+import '../dynamic_layouts.dart';
+
+// The model that tracks the current max size of the Sliver in the mainAxis and
+// tracks if there is still space on the crossAxis.
+class _RunMetrics {
+  _RunMetrics({
+    required this.maxSliver,
+    required this.currentSizeUsed,
+    required this.scrollOffset,
+  });
+
+  double maxSliver;
+  double currentSizeUsed;
+  double scrollOffset;
+}
+
+/// Controls the layout of tiles in a grid.
+/// A [SliverGridLayout] that uses dynamically sized tiles.
+///
+/// Rather that providing a grid with a [SliverGridLayout] directly, you instead
+/// provide the grid a [SliverGridDelegate], which can compute a
+/// [SliverGridLayout] given the current [SliverConstraints].
+///
+/// This layout is used by [SliverGridDelegateWithWrapping].
+///
+/// See also:
+///
+///  * [SliverGridDelegateWithWrapping], which uses this layout.
+///  * [SliverGridLayout], which represents an arbitrary tile layout.
```

Comment:
This can probably be removed since it is an abstract base class users are unlikely to use, it is also not as relevant as the `Dynamic*` version